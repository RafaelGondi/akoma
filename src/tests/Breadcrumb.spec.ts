// @vitest-environment jsdom
import { describe, test, expect } from 'vitest';
import Breadcrumb from '../components/Breadcrumb.vue';
import { mount } from '@vue/test-utils';

const items = [
	{
		label: 'Relat칩rios',
		route: {
			path: '/reports',
			name: 'reports'
		},
	},
	{
		label: 'Relat칩rios individualizados',
		route: {
			path: '/individualized-reports',
			name: 'individualized'
		},
	},
	{
		label: 'Relat칩rio de usu치rios',
		route: {
			path: '/users-reports',
			name: 'users'
		},
	},
];

describe('Breadcrumb snapshot test', () => {
	test('renders correctly', () => {
		const wrapper = mount(Breadcrumb, {
			props: {
				items: items,
			},
		});

		expect(wrapper.html()).toMatchSnapshot();
	});
});
