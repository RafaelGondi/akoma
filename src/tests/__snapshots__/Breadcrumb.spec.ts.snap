// Vitest Snapshot v1

exports[`Breadcrumb snapshot test > renders correctly 1`] = `
"<nav data-v-1a0f46d9=\\"\\">
  <ul class=\\"breadcrumb\\" data-v-1a0f46d9=\\"\\">
    <li role=\\"presentation\\" data-v-1a0f46d9=\\"\\">
      <!--
		Evento emitido quando o wrapper é clicado
		@event cds-click
		@type {Event}
	-->
      <div class=\\"clickable__container--clickable\\" data-v-1a0f46d9=\\"\\">
        <router-link to=\\"/reports\\" class=\\"breadcrumb__link\\" data-v-1a0f46d9=\\"\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"15\\" height=\\"15\\" viewBox=\\"0 0 24 24\\" aria-labelledby=\\"home-outline\\" role=\\"presentation\\" data-v-1a0f46d9=\\"\\">
            <g fill=\\"currentColor\\"></g>
          </svg>
          <div class=\\"spacer\\" data-v-9cc3ee57=\\"\\" data-v-1a0f46d9=\\"\\">Relatórios</div>
        </router-link>
      </div>
    </li>
    <li role=\\"presentation\\" data-v-1a0f46d9=\\"\\">
      <div class=\\"spacer breadcrumb__item\\" data-v-9cc3ee57=\\"\\" data-v-1a0f46d9=\\"\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"16\\" height=\\"16\\" viewBox=\\"0 0 24 24\\" aria-labelledby=\\"caret-right-outline\\" role=\\"presentation\\" class=\\"breadcrumb__separator\\" data-v-1a0f46d9=\\"\\">
          <g fill=\\"currentColor\\"></g>
        </svg>
        <router-link to=\\"/individualized-reports\\" class=\\"breadcrumb__link\\" data-v-1a0f46d9=\\"\\">
          <!--
		Evento emitido quando o wrapper é clicado
		@event cds-click
		@type {Event}
	-->
          <div class=\\"clickable__container--clickable\\" data-v-1a0f46d9=\\"\\">Relatórios individualizados</div>
        </router-link>
      </div>
    </li>
    <li role=\\"presentation\\" data-v-1a0f46d9=\\"\\">
      <div class=\\"spacer breadcrumb__item\\" data-v-9cc3ee57=\\"\\" data-v-1a0f46d9=\\"\\"><svg xmlns=\\"http://www.w3.org/2000/svg\\" width=\\"16\\" height=\\"16\\" viewBox=\\"0 0 24 24\\" aria-labelledby=\\"caret-right-outline\\" role=\\"presentation\\" class=\\"breadcrumb__separator\\" data-v-1a0f46d9=\\"\\">
          <g fill=\\"currentColor\\"></g>
        </svg>
        <router-link to=\\"/users-reports\\" class=\\"breadcrumb__link\\" data-v-1a0f46d9=\\"\\">
          <!--
		Evento emitido quando o wrapper é clicado
		@event cds-click
		@type {Event}
	-->
          <div class=\\"clickable__container--clickable\\" data-v-1a0f46d9=\\"\\">Relatório de usuários</div>
        </router-link>
      </div>
    </li>
  </ul>
</nav>"
`;
